static/js/components/EmploymentForm.js

```diff
@@ -120,9 +120,9 @@ class EmploymentForm extends ProfileFormFields {
     if ( ui.workHistoryEdit === true ) {
       let workHistoryRows = [];
       if ( !_.isUndefined(work_history) ) {
-        workHistoryRows = Object.entries(work_history).filter(([,entry]: [string, WorkHistoryEntry]) =>
-          entry.id !== undefined
-        ).map(([i, entry]) => this.jobRow(entry, i));
+        workHistoryRows = work_history.map((entry, index) => (
```

Comment:
Can you annotate `entry` and `index` here? It looks like flow isn't doing a good job infering the types for these